test/e2e/kubevirt.go

```diff
@@ -287,22 +287,46 @@ var _ = Describe("Kubevirt Virtual Machines", func() {
 			}
 		}
 
-		checkEastWestIperfTraffic = func(vmi *kubevirtv1.VirtualMachineInstance, podIPsByName map[string][]string, stage string) {
+		startEastWestIperfTraffic = func(vmi *kubevirtv1.VirtualMachineInstance, podIPsByName map[string][]string, stage string) error {
 			GinkgoHelper()
 			Expect(podIPsByName).ToNot(BeEmpty())
 			polling := 15 * time.Second
-			timeout := time.Minute
 			for podName, podIPs := range podIPsByName {
 				for _, podIP := range podIPs {
-					output := ""
-					Eventually(func() error {
-						var err error
-						output, err = kubevirt.RunCommand(vmi, fmt.Sprintf("iperf3 -t 1 -c %s", podIP), polling)
-						return err
+					output, err := kubevirt.RunCommand(vmi, fmt.Sprintf("iperf3 -t 0 -c %[2]s --logfile /tmp/%[1]s_%[2]s_iperf3.log &", podName, podIP), polling)
+					if err != nil {
+						return fmt.Errorf("%s: %w", output, err)
+					}
+				}
+			}
+			return nil
+		}
+
+		checkEastWestIperfTraffic = func(vmi *kubevirtv1.VirtualMachineInstance, podIPsByName map[string][]string, stage string) {
+			GinkgoHelper()
+			Expect(podIPsByName).ToNot(BeEmpty())
+			for podName, podIPs := range podIPsByName {
+				for _, podIP := range podIPs {
+					// Check the last line eventually show traffic flowing
+					Eventually(func() (string, error) {
+						iperfLog, err := kubevirt.RunCommand(vmi, fmt.Sprintf("cat /tmp/%s_%s_iperf3.log", podName, podIP), 2*time.Second)
+						if err != nil {
+							return "", err
+						}
+						iperfLogLines := strings.Split(iperfLog, "\n")
+						if len(iperfLogLines) == 0 {
+							return "", nil
+						}
+						return iperfLogLines[len(iperfLogLines)-1], nil
 					}).
-						WithPolling(polling).
-						WithTimeout(timeout).
-						Should(Succeed(), func() string { return stage + ": " + podName + ": " + output })
+						WithPolling(time.Second).
+						WithTimeout(3 * time.Minute).
+						Should(
+							SatisfyAll(
+								ContainSubstring(" sec "),
+								Not(ContainSubstring("0.00 Bytes  0.00 bits/sec")),
```

Comment:
This works for both downtime and tcp connectio broken, so I don't need different mechanisms, also found that sometimes even with "--on-off" the server do not break when tcp connection was broken, for example if it do not receive the packet with wront tuple so it will not abort the tcp connection.

About "0.01 Bytes 0.00 bits/sec" we can fine tune this a little more after merge it.